### PR TITLE
Added support for the astgen macOS arm binary

### PR DESCRIPTION
**Pull request by max-leuthaeuser**

### CODE REVIEW

File: joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/utils/Environment.scala [hunk 1]
```diff
@@ -5,8 +5,9 @@ import java.nio.file.Paths
 
 object Environment {
 
-  val IS_MAC: Boolean   = scala.util.Properties.isMac
-  val IS_LINUX: Boolean = scala.util.Properties.isLinux
+  val IS_MAC: Boolean     = scala.util.Properties.isMac
```
Comment (mpollmeier) [inline joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/utils/Environment.scala:8]: if we change this to `IS_MAC_X86` (or similar) it'd be more specific, and the order of the conditional in `AstGenRUnner` wouldn't matter any more... just a thought

**Review by mpollmeier — APPROVED** [review]

File: joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/utils/Environment.scala (hunk 1 above)
Comment (max-leuthaeuser) [reply joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/utils/Environment.scala]: Yes, right. I changed that to something more explicit. Please have a look :-)

**Review by mpollmeier — APPROVED** [review]
Not sure if this causes any issues, and it's not really important, but if we followed scala's naming convention it would be `val OperatorSystem` and `val Architecture`.

**Comment by max-leuthaeuser** [comment]
Right. The Scala naming convention suggests to use camel case with capital starting letter.
I took the chance to change that everywhere in jssrc2cpg.

**Review by mpollmeier — APPROVED** [review]

**Comment by max-leuthaeuser** [comment]
@mpollmeier I removed the caching and it worked.

**Comment by mpollmeier** [comment]
🎉

**Comment by mpollmeier** [comment]
that might significantly slow down the PR builds though, because it'll download half of maven central every time...
maybe just exclude the ammonite directory from the cache?

**Comment by mpollmeier** [comment]
your test probably did exactly that

**Pull request merged**